runtime/bcutil/cfreader.c

```diff
@@ -3132,8 +3132,8 @@ j9bcutil_readClassFileBytes(J9PortLibrary *portLib,
 	}
 
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
-	/* Currently value type is built on JDK23, so compare with JDK23 for now. */
-	if ((flags & BCT_MajorClassFileVersionMask) < BCT_JavaMajorVersionShifted(23)) {
+	/* Currently value type is built on JDK24, so compare with JDK24 for now. */
+	if ((flags & BCT_MajorClassFileVersionMask) < BCT_JavaMajorVersionShifted(24)) {
```

Comment:
Do you mind updating and using VALUE_TYPES_MAJOR_VERSION and PREVIEW_MINOR_VERSION instead? I created these to try to minimize the number of places that need to be updated during a version change
https://github.com/eclipse-openj9/openj9/blob/master/runtime/oti/j9consts.h#L945-L946

Reply:
Yup, use `J9_IS_CLASSFILE_VALUETYPE()` and also update `VALUE_TYPES_MAJOR_VERSION` to `68`.

Tried [a personal build](https://hyc-runtimes-jenkins.swg-devops.com/job/Pipeline_Build_Test_JDKnext_x86-64_linux_valhalla/1880/)
```
10:16:00  Exception in thread "main" java.lang.ClassFormatError: JVMCFRE190 Fields of value classes must have either ACC_STATIC or ACC_FINAL flags set.; class=org/openj9/test/lworld/NullRestrictedTypeOptTests$TestStoreToNullRestrictedField, offset=785
10:16:00  	at java.base/java.lang.ClassLoader.defineClassImpl(Native Method)
10:16:00  	at java.base/java.lang.ClassLoader.defineClassInternal(ClassLoader.java:470)
10:16:00  	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:431)
10:16:00  	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)
10:16:00  	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:1174)
10:16:00  	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:961)
10:16:00  	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:867)
10:16:00  	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:825)
10:16:00  	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
10:16:00  	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:1099)
10:16:00  	at java.base/java.lang.Class.getDeclaredClassesImpl(Native Method)
10:16:00  	at java.base/java.lang.Class.getClasses(Class.java:605)
10:16:00  	at org.testng.internal.ClassInfoMap.registerClass(ClassInfoMap.java:43)
10:16:00  	at org.testng.internal.ClassInfoMap.<init>(ClassInfoMap.java:29)
10:16:00  	at org.testng.internal.ClassInfoMap.<init>(ClassInfoMap.java:21)
10:16:00  	at org.testng.TestRunner.initMethods(TestRunner.java:369)
10:16:00  	at org.testng.TestRunner.init(TestRunner.java:271)
10:16:00  	at org.testng.TestRunner.init(TestRunner.java:241)
10:16:00  	at org.testng.TestRunner.<init>(TestRunner.java:167)
10:16:00  	at org.testng.SuiteRunner$DefaultTestRunnerFactory.newTestRunner(SuiteRunner.java:663)
10:16:00  	at org.testng.SuiteRunner.init(SuiteRunner.java:260)
10:16:00  	at org.testng.SuiteRunner.<init>(SuiteRunner.java:198)
10:16:00  	at org.testng.TestNG.createSuiteRunner(TestNG.java:1295)
10:16:00  	at org.testng.TestNG.createSuiteRunners(TestNG.java:1273)
10:16:00  	at org.testng.TestNG.runSuitesLocally(TestNG.java:1128)
10:16:00  	at org.testng.TestNG.runSuites(TestNG.java:1049)
10:16:00  	at org.testng.TestNG.run(TestNG.java:1017)
10:16:00  	at org.testng.TestNG.privateMain(TestNG.java:1354)
10:16:00  	at org.testng.TestNG.main(TestNG.java:1323)
10:16:00  -----------------------------------
10:16:00  NullRestrictedTypeOptTests_0_FAILED
```

Is this something related OpenJDK Valhalla update?

Reply:
I will take a look.

Reply:
The check should not be using `J9_IS_CLASSFILE_VALUETYPE` which checks some accessFlags, it should only be checking if the class version is 68 preview.

By using `J9_IS_CLASSFILE_VALUETYPE` the logic is removing the CFR_ACC_IDENTITY flag from classes it shouldn't.

Reply:
Okay, just compare with `VALUE_TYPES_MAJOR_VERSION` which matches the existing comparison with Java major version.

Reply:
Consolidated the checking of `VALUE_TYPES_MAJOR_VERSION` & `PREVIEW_MINOR_VERSION` in `J9_IS_VALUETYPE_VERSION(classfile)`.

Reply:
Looks great, thank you!